src/sklearn_evaluation/plot/pca.py

```diff
@@ -0,0 +1,168 @@
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from sklearn_evaluation.telemetry import SKLearnEvaluationLogger
+from ploomber_core.exceptions import modify_exceptions
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def _set_ax_settings(ax, ind1, ind2, targets=None):
+    ax.set_title("Principal Component Plot", fontsize=10)
+    ax.set_xlabel(f"Principal Component {ind1}", fontsize=10)
+    ax.set_ylabel(f"Principal Component {ind2}", fontsize=10)
+    if targets is not None:
+        ax.legend(targets)
+    ax.grid()
+
+
+def _validate_inputs(X, n_components, target_names, colors, ax):
+    if np.isnan(X).sum():
+        raise ValueError("X array should not consist nan " "pca")
+
+    if n_components < 2:
+        raise ValueError("n_components should be >= 2")
+
+    if target_names and colors and len(target_names) != len(colors):
+        raise ValueError(
+            f"Length received. target_names : {len(target_names)}, "
+            f"colors : {len(colors)}. Lengths should be same"
+        )
+
+    if ax is not None:
+        ax_count = int((n_components * (n_components - 1)) / 2)
+        if len(ax) != ax_count:
+            raise ValueError(f"Number of axes passed should be {ax_count}")
+
+
+@modify_exceptions
+@SKLearnEvaluationLogger.log(feature="plot")
+def pca(X, y=None, target_names=None, n_components=2, colors=None, ax=None):
+    """
+    Plot principle component analysis curve.
+
+    Parameters
+    ----------
+
+    X : array-like, shape = [n_samples, n_features]
+        Training data, where n_samples is the number of samples and
+        n_features is the number of features
+
+    y : array-like or list or None
+        set None if ignored otherwise, pass the targets here
+
+    target_names: list, optional
+        list of target variable names
+
+    n_components : int, float or 'mle', default=None
+        Number of components to keep.
```

Comment:
the default needs updates here, it says None but it's two. also let's add more detail: "if 2, it plots the first and second components, it 3 or more, it plots all pairwise comparisons"